Major refactoring and CI

kostyrev opened the pull request:
I'd like to propose to get rid of version file.
Do you use this file for something else other than fot idempotency?
to review please use https://github.com/inkatze/wildfly/pull/15/files?w=1

kostyrev committed:
Remove version file and refactor install idempotency

kostyrev committed:
Remove version file and refactor install idempotency

kostyrev committed:
Optionally create symlink pointing to installed wildfly version

kostyrev committed:
Optionally create symlink pointing to installed wildfly version

kostyrev committed:
Optionally specify shell for wildfly user

kostyrev committed:
Optionally specify shell for wildfly user

kostyrev committed:
Add optional STARTUP_WAIT and SHUTDOWN_WAIT

kostyrev committed:
Add optional STARTUP_WAIT and SHUTDOWN_WAIT

kostyrev committed:
Add optional jboss.node.name

kostyrev committed:
Add optional jboss.node.name

kostyrev committed:
Optionally point JBOSS_HOME to symlink if wildfly_create_symlink is true

kostyrev committed:
Optionally point JBOSS_HOME to symlink if wildfly_create_symlink is true

kostyrev committed:
Optionally manage firewalld (defaults to true)

kostyrev committed:
Optionally manage firewalld (defaults to true)

kostyrev committed:
Render initd script for CentOS 6

kostyrev committed:
Render initd script for CentOS 6

kostyrev committed:
Remove obsolete tasks

kostyrev committed:
Remove obsolete tasks

kostyrev committed:
Optionally manage standalone data (defaults to true)

kostyrev committed:
Optionally manage standalone data (defaults to true)

kostyrev committed:
Optionally remove wildfly tar.gz

kostyrev committed:
Optionally remove wildfly tar.gz

inkatze commented:
The version file has two purposes: avoid reinstalling the same version unless said otherwise (which apparently I forgot to add to this role) and idempotency which will be useful once I figure out how to add CI to the project.

I could merge your PR since the existence of the version file is pointless right now, unless you value idempotency, but I'd like to now if it'd make more sense if I added it again later on for the reasons listed above.

juhoffma committed:
Fixed another typo (#14)

* Fixed a typo

* Fixes #8

A systemd definition requires the identification of a given Service, whether it is a service or a socket, etc. This bugfix fixes #8. But there are more things to do. The init script allowed for a lot more options, which need to be ported into the systemd script. I think the easiest would be to call the init script with the start option from the service definition file. Let me know what you think.

kostyrev commented:
With this PR you avoid reinstalling the same version unless you delete manually /opt/wildfly-version dir
In the future we can add a task that deletes {{ wildfly_install_dir }} when some toggle is true.
This PR provides the idempotency.
I can send you a PR with [molecule](https://molecule.readthedocs.io) and Travis CI if you are interested.
See [this role](https://github.com/kostyrevaa/ansible-role-epel) for example

inkatze commented:
I see. I would really appreciate a PR with molecule and Travis CI. And about this one, I think I can test it and merge it in a little while.

Thanks.

kostyrev committed:
Optionally create mgmt users and app users

kostyrev committed:
Optionally create mgmt users and app users

kostyrev committed:
Add optional wildfly_bind_address_unsecure and wildfly_messaging_group_address

kostyrev committed:
Add optional wildfly_bind_address_unsecure and wildfly_messaging_group_address

kostyrev committed:
Optionally manage service (defaults to true)

kostyrev committed:
Optionally manage service (defaults to true)

kostyrev committed:
Do not log passwords of users

kostyrev committed:
Do not log passwords of users

kostyrev committed:
Add molecule for role testing

kostyrev committed:
Add molecule for role testing

kostyrev committed:
Do not mess with firewalld on CentOS 6

kostyrev committed:
Do not mess with firewalld on CentOS 6

kostyrev committed:
Add Travis

kostyrev committed:
Add Travis

kostyrev committed:
Install tested version of molecule in Travis

kostyrev committed:
Install tested version of molecule in Travis

inkatze commented on playbook.yml:
```diff
@@ -0,0 +1,11 @@
+---
+- hosts: all
```
Is it ok for the playbook not to have `become: yes`?

kostyrev replied:
It's not necessary.
BTW, I fixed molecule and Travis issue.
See [this](https://travis-ci.org/kostyrevaa/wildfly/builds/180369421) successful Travis build.

inkatze commented:
Quick update: I still have to dig a little bit more into the refactoring and test it with the init service manager but is looking good so far.

kostyrev committed:
Fix molecule in Travis

kostyrev committed:
Fix molecule in Travis

kostyrev commented:
I used initd as well.

kostyrev committed:
Add EL6 to meta/main.yml

kostyrev committed:
Merge branch 'master' of github.com:kostyrevaa/wildfly

kostyrev committed:
Add CentOS7 to molecule

kostyrev commented:
Yet here are some other changes. I rebased my changes with latest changes in the role.
See this [build log](https://travis-ci.org/kostyrevaa/wildfly/builds/180401958)

inkatze commented:
Everything looks great! I see you took care of a lot of use cases I never considered. 

Just one small thing though, I think we should stick with one format for boolean values. By the time I created this role the most widespread format was True/False. I think Yes/No looks nicer in YAML, but I don't have any particular preference for either and I'm not sure of what's the preferred format in the community. What do you think?

inkatze commented:
Created #20 to deal with the boolean situation.

inkatze commented:
Fixes #17

inkatze commented:
Resolves #17

inkatze commented:
Thanks for your great contribution! Having CI will help a lot in the future and users will be happy for the improved running time.

kostyrev commented:
About CI
You need to configure CI for your github account.
Now the Travis build badge (in README.md) points to my account.
Just follow [getting started](https://docs.travis-ci.com/user/getting-started/#To-get-started-with-Travis-CI%3A)

inkatze commented:
I was so excited watching the builds working that I completely forgot about the badge.

---
Juan Diego Romero González

On Dec 2, 2016, 3:41 AM -0600, Александр Костырев <notifications@github.com>, wrote:
>
> About CI
> You need to configure CI for your github account.
> Now the Travis build badge is pointed to my account.
> Just follow getting started (https://docs.travis-ci.com/user/getting-started/#To-get-started-with-Travis-CI%3A)
>
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub (https://github.com/inkatze/wildfly/pull/15#issuecomment-264413455), or mute the thread (https://github.com/notifications/unsubscribe-auth/AB721ir87c2gE_3ciJlmDDfE-Z6hoUfJks5rD-fPgaJpZM4K_RSf).
>
>
>
